src/test/datascience/notebook/notebookEditor.vscode.test.ts

```diff
@@ -115,4 +121,32 @@ suite('Notebook Editor tests', () => {
         // The third cell should have a runState of Success
         assert.strictEqual(thirdCell?.metadata.runState, vscodeNotebookEnums.NotebookCellRunState.Success);
     });
+
+    test('Switch kernels', async function () {
+        // add a cell
+        await insertCodeCell('print("0")', { index: 0 });
+
+        const cell = vscodeNotebook.activeNotebookEditor?.document.cells![0]!;
+
+        await executeCell(cell);
+
+        // Wait till execution count changes and status is success.
+        await waitForExecutionCompletedSuccessfully(cell);
+
+        // Switch kernels to the other kernel
+        const kernels = await kernelProvider.provideKernels(
+            vscodeNotebook.activeNotebookEditor!.document,
+            CancellationToken.None
+        );
+        traceInfo(`Kernels found for switch kernel: ${kernels?.map((k) => k.label).join('\n')}`);
+        if (kernels?.length && kernels?.length > 1) {
+            // We have multiple kernels. Try switching
+            await waitForKernelToGetSelected(kernels[1].label);
+        }
+
+        // Execute cell and verify output
+        await executeCell(cell);
```

Comment:
This doesn't verify the cell was selected.
Shouldnt we check `sys.executable`.
In the past we have had issues where kernel selection says one thing & the python environment under which things are running is different.
I'd prefer a more robust test to check `sys.exeutable`

Reply:
Also guards against conda kernel selection. Where users aren't able to select conda kernels.
This way we also test changing a kernel & also executing against that kernel.

Reply:
THe problem would be verifying what path was selected. Hard to predict what it should be.

Reply:
Sorry I don't think we should change it to check the sys path. Too hard to get it to come out correct. I can't predict which kernel it's going to install and then there's a dependency between the kernels we have in main.yml and this code.

However we can determine if it's different though. I'll add that.

Reply:
Yea, checking if its different is better if we cannot determine the kernel.
At least in the case of remote Jupyter it will not work with the tests.